generator/uscore/metadata_extractor.rb

```diff
@@ -0,0 +1,251 @@
+# frozen_string_literal: true
+
+module Inferno
+  module Generator
+    module USCoreMetadataExtractor
+      PROFILE_URIS = Inferno::ValidationUtil::US_CORE_R4_URIS
+
+      def profile_uri(profile)
+        "http://hl7.org/fhir/us/core/StructureDefinition/#{profile}"
+      end
+
+      def profile_json_uri(profile)
+        "https://www.hl7.org/fhir/us/core/StructureDefinition-#{profile}.json"
+      end
+
+      def search_param_path(resource, param)
+        param = 'id' if param == '_id'
+        "SearchParameter/us-core-#{resource.downcase}-#{param}"
+      end
+
+      def extract_metadata
+        metadata = {
+          # The 'key' for the module is just the directory the IG is in
+          name: @path
+        }
+
+        # Note: consider using the Ruby representation instead of JSON
+        # We aren't right now because some information is being scrubbed
+        # from SearchParameter resource
+
+        capability_statement_json = capability_statement('server')
+        add_metadata_from_ig(metadata, ig_resource)
+        add_metadata_from_resources(metadata, capability_statement_json['rest'][0]['resource'])
+        add_special_cases(metadata)
+      end
+
+      def add_metadata_from_ig(metadata, implementation_guide)
+        metadata[:title] = "#{implementation_guide['title']} v#{implementation_guide['version']}"
+        metadata[:description] = "#{implementation_guide['title']} v#{implementation_guide['version']}"
+      end
+
+      def build_new_sequence(resource, profile)
+        base_path = profile.split('us/core/').last
+        base_name = profile.split('StructureDefinition/').last
+        profile_json = @resource_by_path[base_path]
+        reformatted_version = ig_resource['version'].delete('.')
+        profile_title = profile_json['title'].gsub(/US\s*Core\s*/, '').gsub(/\s*Profile/, '').strip
+        class_name = base_name
+          .split('-')
+          .map(&:capitalize)
+          .join
+          .gsub('UsCore', "USCore#{reformatted_version}") + 'Sequence'
+
+        # In case the profile doesn't start with US Core
+        class_name = "USCore#{reformatted_version}#{class_name}" unless class_name.start_with? 'USCore'
+
+        {
+          name: base_name.tr('-', '_'),
+          class_name: class_name,
+          resource: resource['type'],
+          profile: profile_uri(base_name), # link in capability statement is incorrect,
+          profile_json: profile_json_uri(base_name),
```

Comment:
You can delete this entire line.

Reply:
Yup, removed this as well, if we don't ever need this.